github3/pulls.py

```diff
@@ -286,6 +286,49 @@ def create_review_comment(self, body, commit_id, path, position):
         json = self._json(self._post(url, data=data), 201)
         return self._instance_or_null(ReviewComment, json)
 
+    @requires_auth
+    def create_review(self, body, commit_id=None, event=None,
+                      comments=None):
+        """Create a review comment on this pull request.
```

Comment:
Something I think I learned from the tests is that this method always creates a "pending" review and that you then need to call "submit" on the returned `PullReview` instance. Am I understanding that correctly?

If so, I'm wondering if we can make the method name and doc-string clearer on that topic. Something, perhaps, like

```py
def create_pending_review(...):
    """Create a pending review on this pull request.

    This will not finalize the review left on the pull request. This will, however,
    return the :class:`~github3.pulls.PullReview` and to finalize the review, you must
    call :meth:`~github3.pulls.PullReview.submit`.

    :param str body:
        ...
    """
```

Reply:
Depends on whether 'event' is left at `None` or set to `'APPROVE'`, `'REQUEST_CHANGES'` or `'COMMENT'`. If set to any of these, the review will be created immediately. If left to `None` it will be created in a `'PENDING'` state and will require calling submit on the returned review to complete.

Have mentioned below on the `event` attribute documentation about this, but maybe it's worth spelling it out at the top again?

Reply:
Meant to include, if you see the test for this method uses 'COMMENT' which created the review comment on this change as part of recording for the integration tests (https://github.com/sigmavirus24/github3.py/pull/819/files#diff-adec5b4785431b0741c97851e7e14dfaR61) 

But for the submit test at https://github.com/sigmavirus24/github3.py/pull/819/files#diff-adec5b4785431b0741c97851e7e14dfaR174 I first create the review without setting the `event` param https://github.com/sigmavirus24/github3.py/pull/819/files#diff-adec5b4785431b0741c97851e7e14dfaR179 which shows this second behaviour and then submit it. Course if you fail to set the `event` value for the submit call it will be left in the `PENDING` state anyway.

Reply:
Perhaps the following is reasonable?

```
def create_review(...):
    """Create a review comment on this pull request.

    Leaving the `event` param set to the default of `None` will not finalize the review
    on the pull request. This will still return the :class:`~github3.pulls.PullReview` without
    `submitted_at` set and to finalize/publish the review you must call
    :meth:`~github3.pulls.PullReview.submit`.
    
```

Interestingly although the documentation states this should result in a return code of 422, the code below that I added worked for the integration test without needing to handle 422 instead of 200. Is that normal?